src/upnp_xml.cc

```diff
@@ -485,17 +485,17 @@ std::pair<std::string, bool> UpnpXMLBuilder::renderSubtitle(const std::string& v
 
 std::string UpnpXMLBuilder::renderExtension(const std::string& contentType, const fs::path& location)
 {
-    auto urlExt = URLUtils::joinUrl({ URL_FILE_EXTENSION, "file." });
+    auto urlExt = URLUtils::joinUrl({ URL_FILE_EXTENSION, "file" });
 
     if (!contentType.empty() && (contentType != CONTENT_TYPE_PLAYLIST)) {
-        return fmt::format("{}{}", urlExt, contentType);
+        return fmt::format("{}{}{}", urlExt, ".", contentType);
     }
 
     if (!location.empty() && location.has_extension()) {
         // make sure that the filename does not contain the separator character
-        std::string filename = urlEscape(location.filename().stem().string());
+        // std::string filename = urlEscape(location.filename().stem().string());
         std::string extension = location.filename().extension();
-        return fmt::format("{}{}{}", urlExt, filename, extension);
+        return fmt::format("{}{}", urlExt, extension);
```

Comment:
We need to implement this as quirks because the filename is used as language in e.g. vlc.

Just leave it out of the PR.